cpp/src/slice2matlab/Main.cpp

```diff
@@ -933,6 +931,102 @@ namespace
         }
     }
 
+    void declareArgument(IceInternal::Output& out, const ParameterPtr& param)
+    {
+        out << nl << param->mappedName();
+
+        TypePtr type = param->type();
+
+        // First the dimensions
+
+        bool mustBeScalarOrEmpty = false;
+        if (auto builtin = dynamic_pointer_cast<Builtin>(type))
+        {
+            if (builtin->kind() < Builtin::KindString)
+            {
+                out << " (1, 1)";
+            }
+            else if (builtin->kind() == Builtin::KindString)
+            {
+                out << " (1, :)";
+            }
+            else // ObjectPrx and Value
+            {
+                mustBeScalarOrEmpty = true;
+            }
+        }
+        else if (
+            dynamic_pointer_cast<Enum>(type) || dynamic_pointer_cast<Dictionary>(type) ||
+            dynamic_pointer_cast<Struct>(type))
+        {
+            out << " (1, 1)";
+        }
+        else if (dynamic_pointer_cast<Sequence>(type))
+        {
+            out << " (1, :)";
+        }
+        else // proxies and classes
+        {
+            mustBeScalarOrEmpty = true;
+        }
+
+        // We can't specify a type for optional parameter because we can't represent "not set" with the same MATLAB
+        // type.
+        if (!param->optional())
```

Comment:
It's more comprehensive that the Property typing since there is no "convert" issue here.